src/Npgsql/Internal/NpgsqlConnector.cs

```diff
@@ -711,7 +715,11 @@ void WriteStartupMessage(string username)
         WriteStartup(startupParams);
     }
 
+#if NET5_0_OR_GREATER
+    async ValueTask<string> GetUsernameAsync()
```

Comment:
We can do the usual pattern of accepting an async bool parameter and checking that to decide whether to call the sync or async overload below.

BTW should we be passing a cancellation token as well?